pkg/extension/extension.go

```diff
@@ -0,0 +1,22 @@
+package extension
+
+import (
+	"context"
+
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Extension interface{}
```

Comment:
I think I also don't get the motivation for the flexibility. What "new extension points" are you thinking of? The logic of the reconciler is really fairly simple, and the Helm-related actions should remain atomic from an extension point of view.

Reply:
I like the idea with using the strongly typed registration.

Reply:
Apropos "empty interface" -- I have actually been wondering if some `Name() string` method should be added to the `Extension` interface so that, for example, errors from the extension can be logged together with the failing extensions' name. WDYT?

Reply:
Regarding the "new extension points", this is actually something that Joe brought up. I have asked him about this, though.